plugins/inputs/t128_tank/reader.go

```diff
@@ -285,14 +289,7 @@ parseLoop:
 		case r.sendChan <- collectedMessages:
 		}
 	}
-	if cmdErr := cmd.Wait(); cmdErr != nil {
-		var exitErr *exec.ExitError
-		if errors.As(cmdErr, &exitErr) {
-			r.log.Errorf("%s tank read command exited with error: %s", r.topic, exitErr.Error())
-		} else {
-			r.log.Errorf("%s tank read command exited with error: %s", r.topic, cmdErr)
-		}
-	}
+	r.handleCmdExitError(cmd)
```

Comment:
It really seems like we could just this in a single defer rather than catching each codepath separately.
```go
defer func() {
	cmdContextCancel()
	if cmdErr := cmd.Wait(); cmdErr != nil {
		var exitErr *exec.ExitError
		if errors.As(cmdErr, &exitErr) {
			r.log.Errorf("%s tank read command exited with error: %s", r.topic, exitErr.Error())
		} else {
			r.log.Errorf("%s tank read command exited with error: %s", r.topic, cmdErr)
		}
	}
}()
```